### PR TITLE
Lets Fame - Feb code changes and additional implements

### PR DESCRIPTION
**Pull request by Ansariac**

### CODE REVIEW

File: src/main/java/com/letsfame/Req/LetsFameOrderReq.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package com.letsfame.Req;
```
Comment (Pikazza) [inline src/main/java/com/letsfame/Req/LetsFameOrderReq.java:1]: package name is not updated, change it to 

com.letsfame.request;

File: src/main/java/com/letsfame/Req/LetsFameOrderReq.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class LetsFameOrderReq {
```
Comment (Pikazza) [inline src/main/java/com/letsfame/Req/LetsFameOrderReq.java:5]: Rename the class from LetsFameOrderReq to either OrderCreateRequest or OrderUpdateRequest based on the operation

**!Important!** remove all LetsFame from all requests and model class

File: src/main/java/com/letsfame/advice/CustomExceptionHandler.java [hunk 3]
```diff
@@ -63,6 +65,7 @@ public ResponseEntity<?> handleInvalidObjectException(Exception ex, WebRequest r
 	protected ResponseEntity<Object> handleHttpMessageNotReadable(HttpMessageNotReadableException ex,
 			HttpHeaders headers, HttpStatus status, WebRequest request) {
 		log.info("Handling handleHttpMessageNotReadable Exception");
+		log.info("Exception:"+ExceptionUtils.getStackTrace(ex));
```
Comment (Pikazza) [inline src/main/java/com/letsfame/advice/CustomExceptionHandler.java:68]: merge both loggers into a single one, there is no need of having two loggers

File: src/main/java/com/letsfame/bean/LetsFameOrder.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
+
+@Getter
+@Setter
+@ToString
+@Document(collection = "Order")
+public class LetsFameOrder implements Serializable {
```
Comment (Pikazza) [inline src/main/java/com/letsfame/bean/LetsFameOrder.java:16]: rename class name to Order

**Important** follow the same for remaining classes

File: src/main/java/com/letsfame/config/AppConfig.java [hunk 5]
```diff
@@ -9,7 +9,7 @@
 
 @Configuration
 @EnableWebMvc
```
Comment (Pikazza) [inline src/main/java/com/letsfame/config/AppConfig.java:11]: why you have used EnableWebMvc annotation here,

can we have a common config class for all the beans?

File: src/main/java/com/letsfame/config/SwaggerConfig.java [hunk 6]
```diff
@@ -18,28 +18,18 @@
 @EnableSwagger2
 @EnableAutoConfiguration
```
Comment (Pikazza) [inline src/main/java/com/letsfame/config/SwaggerConfig.java:19]: why do we have EnableAutoConfiguration annotation here?

File: src/main/java/com/letsfame/config/SwaggerConfig.java [hunk 7]
```diff
@@ -18,28 +18,18 @@
...
 
-public class SwaggerConfig  extends WebMvcConfigurationSupport {
-
+public class SwaggerConfig extends WebMvcConfigurationSupport {
```
Comment (Pikazza) [inline src/main/java/com/letsfame/config/SwaggerConfig.java:21]: why WebMvcConfigurationSupport is extended and what s the use of it

File: src/main/java/com/letsfame/config/SwaggerConfig.java [hunk 8]
```diff
@@ -18,28 +18,18 @@
...
```
Comment (Pikazza) [inline src/main/java/com/letsfame/config/SwaggerConfig.java:22]: have a common config for all kind of bean initialization

File: src/main/java/com/letsfame/config/SwaggerConfig.java [hunk 9]
```diff
@@ -18,28 +18,18 @@
 @EnableSwagger2
```
Comment (Pikazza) [inline src/main/java/com/letsfame/config/SwaggerConfig.java:18]: Springboot supports Swagger, why have you used springfox external library for the swagger?

please use Springboot library

File: src/main/java/com/letsfame/controller/OrderController.java [hunk 10]
```diff
@@ -1,61 +1,60 @@
...
+import com.letsfame.bean.LetsFameOrder;
+import com.letsfame.response.ResponseHandler;
+import com.letsfame.service.OrderService;
+import com.razorpay.Order;
+
+import io.swagger.annotations.ApiOperation;
+
+@RestController
+@RequestMapping("/txn_api/v1.0/order")
+public class OrderController {
```
Comment (Pikazza) [inline src/main/java/com/letsfame/controller/OrderController.java:25]: What is the use of OrderContoller and Payment Controller and Subscription controller? why do we need all these three?

post this info in the group

File: src/main/java/com/letsfame/bean/WebhookPaymentDetails.java [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
+
+@Getter
+@Setter
+@ToString
+@Document(collection = "Webhook")
```
Comment (Pikazza) [inline src/main/java/com/letsfame/bean/WebhookPaymentDetails.java:17]: there is no need of creating Webhook collection, we not all need to store them in our database inself

File: src/main/java/com/letsfame/dto/SubscriptionsDto.java [hunk 12]
```diff
@@ -3,27 +3,22 @@
...
+import lombok.ToString;
 
 @Getter
 @Setter
-@NoArgsConstructor
-@AllArgsConstructor
-//@Document(collection = "subscription")
+@ToString
+
 public class SubscriptionsDto implements Serializable {
```
Comment (Pikazza) [inline src/main/java/com/letsfame/dto/SubscriptionsDto.java:18]: what is the use of dto package and repository package? cant we have it single package

explain this in group?

File: src/main/java/com/letsfame/serviceImpl/WebhookServiceImpl.java [hunk 13]
```diff
@@ -1,39 +1,39 @@
...
 import com.razorpay.RazorpayClient;
 
 @Service
 @Transactional
-public class WebhookServiceImpl implements webhookService {
+public class WebhookServiceImpl implements WebhookService {
 
 	@Autowired
-	private WebHooksResponseRepository webHooksRepository;
+	private WebhooksRepository webHooksRepository;
```
Comment (Pikazza) [inline src/main/java/com/letsfame/serviceImpl/WebhookServiceImpl.java:36]: there is no need of storing them in the Database itself please remove this. if you ahve more questions please reach me

File: src/main/java/com/letsfame/advice/CustomExceptionHandler.java (hunk 3 above)
Comment (Silaluthirapathi) [reply src/main/java/com/letsfame/advice/CustomExceptionHandler.java:68]: Errors are logged as ERROR not INFO and Check about parametrized logger and implement the same.

File: src/main/java/com/letsfame/Req/LetsFamePlanItemReq.java [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+	@JsonProperty("name")
+	private String name;
+
+	@JsonProperty("amount")
+	private int amount;
+
+	@JsonProperty("currency")
+	private String currency;
+
+	@JsonProperty("description")
```
Comment (Silaluthirapathi) [inline src/main/java/com/letsfame/Req/LetsFamePlanItemReq.java:16]: Here @jsonproperty necessity is not required here. why do you use the annotation? Please understand the purpose of jsonproperty annotation and use it.

@Pikazza Please have a look on this?

File: src/main/java/com/letsfame/Req/LetsFameOrderReq.java [hunk 15]
```diff
@@ -0,0 +1,16 @@
...
+
+	@JsonProperty("amount")
+	private String amount;
+
+	@JsonProperty("currency")
+	private String currency;
```
[the remark this answers is not in the document]
Comment (Silaluthirapathi) [reply src/main/java/com/letsfame/Req/LetsFameOrderReq.java:11]: @Pikazza Please provide your comment on this?

**Review by Silaluthirapathi — CHANGES_REQUESTED** [review]
@sathishkumar144625 , @Pikazza added few comments. Please address those comments.

File: src/main/java/com/letsfame/serviceImpl/WebhookServiceImpl.java (hunk 13 above)
Comment (Pikazza) [reply src/main/java/com/letsfame/serviceImpl/WebhookServiceImpl.java:39]: Remove WebhooksRepository Autowiring itself

File: src/main/java/com/letsfame/serviceImpl/WebhookServiceImpl.java [hunk 16]
```diff
@@ -46,58 +46,65 @@ public class WebhookServiceImpl implements webhookService {
...
 		}
 
-		PaymentDetailsWebhookDto savedData1 = new PaymentDetailsWebhookDto();
-		PaymentStatusByMemberDto paymentstatus = new PaymentStatusByMemberDto();
-		List<String> error = new ArrayList<>();
-		SimpleDateFormat sdf = new SimpleDateFormat("dd/MM/yyyy");
+		WebhookPaymentDetails savedData1 = new WebhookPaymentDetails();
+		WebhookPaymentStatusByMember paymentstatus = new WebhookPaymentStatusByMember();
 
 		System.out.println("webhook ::" + notification);
```
Comment (Pikazza) [inline src/main/java/com/letsfame/serviceImpl/WebhookServiceImpl.java:72]: please change it to logger

File: src/main/java/com/letsfame/serviceImpl/WebhookServiceImpl.java [hunk 17]
```diff
@@ -46,58 +46,65 @@ public class WebhookServiceImpl implements webhookService {
...
 		Response response = new Response();
 		if (!"payment.captured".equalsIgnoreCase(notification.getEvent().getEvent())) {
 			return response;
 		}
 
-		PaymentDetailsWebhookDto savedData1 = new PaymentDetailsWebhookDto();
-		PaymentStatusByMemberDto paymentstatus = new PaymentStatusByMemberDto();
-		List<String> error = new ArrayList<>();
-		SimpleDateFormat sdf = new SimpleDateFormat("dd/MM/yyyy");
+		WebhookPaymentDetails savedData1 = new WebhookPaymentDetails();
```
Comment (Pikazza) [inline src/main/java/com/letsfame/serviceImpl/WebhookServiceImpl.java:69]: Webhook event is common for both Payment and subscriptions. so please change the name WebhookPaymentDetails to a generic name like WebhookEvent

File: src/main/java/com/letsfame/serviceImpl/WebhookServiceImpl.java [hunk 18]
```diff
@@ -46,58 +46,65 @@ public class WebhookServiceImpl implements webhookService {
...
 
-			// To get invoice details for find subscription ID
-			Invoice invoice = razorpayClient.invoices
-					.fetch(notification.getEvent().getPayload().getPayment().getEntity().getInvoice_id());
+		// To get invoice details for find subscription ID
+		Invoice invoice = razorpayClient.invoices
+				.fetch(notification.getEvent().getPayload().getPayment().getEntity().getInvoice_id());
 
-			System.out.println("invoice_details::" + invoice);
+		System.out.println("invoice_details::" + invoice);
```
Comment (Pikazza) [inline src/main/java/com/letsfame/serviceImpl/WebhookServiceImpl.java:78]: change it to logger

File: src/main/java/com/letsfame/serviceImpl/WebhookServiceImpl.java [hunk 19]
```diff
@@ -46,58 +46,65 @@ public class WebhookServiceImpl implements webhookService {
...
 
-			System.out.println("invoice::" + invoice.get("subscription_id"));
-			notification.setSubscriptionId(invoice.get("subscription_id"));
+		System.out.println("invoice::" + invoice.get("subscription_id"));
+		notification.setSubscriptionId(invoice.get("subscription_id"));
 
-//			System.out.println("invoice::" + invoice.get("expire_by"));
-//			notification.setExpireBy(invoice.get("expire_by"));
+		LetsFamePayment payments = paymentService
```
Comment (Pikazza) [inline src/main/java/com/letsfame/serviceImpl/WebhookServiceImpl.java:83]: there is no need of having plural name for a single payment

File: src/main/java/com/letsfame/serviceImpl/WebhookServiceImpl.java [hunk 20]
```diff
@@ -46,58 +46,65 @@ public class WebhookServiceImpl implements webhookService {
...
+				.findByPaymentId(notification.getEvent().getPayload().getPayment().getEntity().getPaymentId());
 
-			// To save payments status
+		// To save payments status
+		if (payments != null) {
 			savedData1 = webHooksRepository.save(notification);
-
+			System.out.println("Payments status updated Sucessfully ::" + savedData1);
```
Comment (Pikazza) [inline src/main/java/com/letsfame/serviceImpl/WebhookServiceImpl.java]: 1. please change it to the logger
2. please remove saving  mehanism

File: src/main/java/com/letsfame/serviceImpl/WebhookServiceImpl.java [hunk 21]
```diff
@@ -46,58 +46,65 @@ public class WebhookServiceImpl implements webhookService {
...
 		if (!"payment.captured".equalsIgnoreCase(notification.getEvent().getEvent())) {
 			return response;
 		}
 
-		PaymentDetailsWebhookDto savedData1 = new PaymentDetailsWebhookDto();
-		PaymentStatusByMemberDto paymentstatus = new PaymentStatusByMemberDto();
-		List<String> error = new ArrayList<>();
-		SimpleDateFormat sdf = new SimpleDateFormat("dd/MM/yyyy");
+		WebhookPaymentDetails savedData1 = new WebhookPaymentDetails();
+		WebhookPaymentStatusByMember paymentstatus = new WebhookPaymentStatusByMember();
```
Comment (Pikazza) [inline src/main/java/com/letsfame/serviceImpl/WebhookServiceImpl.java:70]: change the class name WebhookPaymentStatusByMember

**Review by Pikazza — APPROVED** [review]

**Review by Pikazza — APPROVED** [review]

**Review by Pikazza — APPROVED** [review]

**Review by arulselvamatv — APPROVED** [review]
approved

**Review by Pikazza — APPROVED** [review]
code looking fine

**Review by arulselvamatv — APPROVED** [review]

**Review by arulselvamatv — APPROVED** [review]
approve

**Review by Silaluthirapathi — APPROVED** [review]
LGTM

**Pull request merged**